demo/docbot/docbot.py

```diff
@@ -0,0 +1,64 @@
+import sys
+from tkinter.filedialog import Open
+from token import OP
+from unittest import result
+from opensearchpy import OpenSearch
+sys.path.append('./demo/')
+
+from docbot.language_pipeline import generate_response
+from opensearchpy import OpenSearch
+
+class DocBot():
+  def __init__ (self, open_search:OpenSearch):
+      self.conversations = {}
+      self.OpenSearch = open_search
+
+  def handle_message(self, message_object):
+    # get conversation id form _conversation handler
+    id = self.conversation_handler(message_object.name)
+    # with conversation id call docbot.generate_response(conversation_id, message) and return response
+    return generate_response(id, message_object)
+
+  def conversation_handler(self, conversation_name):
+    # check if conversation in state
+    id = None
+    if conversation_name in self.conversations['name']:
+      # check if conversation exists. If it exists track conversation id in self.conversations by name: id
+      id = self.conversations[conversation_name]
+    else:
+      # Check if it exists in OpenSearch
+      search_query = {
+        "query": {
+            "match": {
+                "name": conversation_name
+                }
+              }
+            }
+      result = self.OpenSearch.search(index='cohere', body=search_query)
+      if result['hits']['total']['value'] > 0:
+        data = result.json()
+        hits = data.get('hits', {}).get('hits', [])
+        id = hits[0]['id']
+      else:
+        max_id_query = {
+          "size": 0,
+          "aggs": {
+            "max_id": {
+              "max": {
+                  "field": "id"
+                }
+              }
+            }
+          }
+        max_id_response = self.OpenSearch.search(index="your_index_name", body=max_id_query)
+        if max_id_response["aggregations"]["max_id"]["value"] != None:
+          max_id = max_id_response["aggregations"]["max_id"]["value"]
+        else:
+          max_id = 0
+        id = max_id + 1
```

Comment:
So this is really clever but we don't need to solve this part in our code. Conversation ID's are automatically generated hash's that are a result of creating a new conversation in OpenSearch. Here if the conversation isn't in our local cache and not in OpenSearch, we can use the below endpoint to create a new conversation

```json
POST /_plugins/_ml/memory/conversation
{
  "name": Example conversation
}
```

This will return a conversation_id such as: { "conversation_id": "4of2c9nhoIuhcr" } which we can then use for conversing.

Reply:
How would I access this in OpenSearch? I tried looking for this in the Python Documentation but I can't seem to find the part of the codebase that would send this POST.

Reply:
We may need to add an API for this in our MonkeyPatch class. Check #97 where Arturo had to do something similar for search pipelines. We can add it there before contributing it upstream to the python client.